sonic_installer/main.py

```diff
@@ -208,6 +208,39 @@ def print_deprecation_warning(deprecated_cmd_or_subcmd, new_cmd_or_subcmd):
                 fg="red", err=True)
     click.secho("Please use '{}' instead".format(new_cmd_or_subcmd), fg="red", err=True)
 
+def run_cmd_or_abort(click, argv):
+    cmd_process = subprocess.Popen(argv, stdout=subprocess.PIPE)
+    stdout, stderr = cmd_process.communicate()
+    if not stdout or cmd_process.returncode:
+        click.echo("Failed to run command '{0}'".format(argv))
+        raise click.Abort()
+
+    return stdout
+
+def update_sonic_environment(click, binary_image_version):
+    sonic_env_template_file = "/usr/share/sonic/templates/sonic-environment.j2"
```

Comment:
>/usr/share/sonic/templates/sonic-environment.j2 [](start = 31, length = 47)

This is template in old image. If future image has new content in this file, how do you render the template?

Reply:
Good point. Those envvars are relatively immutable. new/existing variable can always be retrieved using sonic-cfggen or via batching scripts.

Reply:
Please let me know your concerns. The gain from this change is ~7% for exiting variable. If I addressed your comment, can you please unblock my PR from not being merged.

Reply:
Please help explore how to extract the correct template from the new image and render with sonic-cfggen in old image.

Please also be careful about the error handling if anything broken because of data incompatibility.

---
In reply to: [462628227](https://github.com/Azure/sonic-utilities/pull/1011#discussion_r462628227) [](ancestors = 462628227)

Reply:
The incoming image has its file system compressed and so it is not available during installation time. Those variable are relatively immutable and address starting dockers delays. Adding new variable is low priority for now.

@yxieca is it a known practice to inflate the incoming image in order to retrieve a template file?

@qiluo-msft , @lguohan should this pr be blocked on this request?

Reply:
Yes, you could do that. We do that for warm reboot pre-boot check too.

Reply:
ACK, update the pr.